### PR TITLE
[DPE-5544] - fix: set dir permissions & remove lost+found on storageattached

### PR DESCRIPTION
**Pull request by marcoppenheimer**

### CODE REVIEW

File: src/events/broker.py [hunk 1]
```diff
@@ -350,6 +351,14 @@ def _on_storage_attached(self, event: StorageAttachedEvent) -> None:
                 ]
             )
 
+        if self.charm.substrate == "k8s":
+            self.workload.exec(
+                ["chown", "-R", "kafka:root", f"{self.workload.paths.data_path}/{STORAGE}"]
```
Comment (Batalex) [inline src/events/broker.py:356]: **nitpick, non-blocking**: If you need to revisit the PR, could we use the literal for the user?

Also **question**: We usually use the same `USER:GROUP` everywhere, is the `root` group necessary?

**Review by Batalex — APPROVED** [review]

File: src/events/broker.py (hunk 1 above)
Comment (marcoppenheimer) [reply src/events/broker.py]: Great point, I'm an idiot, especially after having just commented on your current PR the same thing.
Have updated.

**Review by deusebio — APPROVED** [review]
LGTM

**Comment by deusebio** [comment]
I have tested the fix on Azure AKS and the deployment worked fine. Attached the list of revisions tested:

```
App            Version  Status   Scale  Charm            Channel      Rev  Address       Exposed  Message
admin                   unknown      0  data-integrator  latest/edge   72  10.0.96.101   no
consumer                unknown      0  kafka-test-app   latest/edge   13  10.0.15.211   no           
kafka-k8s      3.6.1    waiting    3/0  kafka-k8s                       0  10.0.69.71    no       
producer                unknown      0  kafka-test-app   latest/edge   13  10.0.139.134  no
zookeeper-k8s  3.9.2    active     1/0  zookeeper-k8s    3/edge        67  10.0.124.149  no   
```

(the controller was scaling down, hence the 0 scale units, but the process of producing and consuming worked fine, believe me!)

**Comment by joalif** [comment]
Hi @marcoppenheimer is there an ETA for this to be merged and released? I believe this fixes a bug a customer hits.
Issue: When deploying kafka-k8s on microk8s with ceph backend the kafka unit and application get into blocked state.
This does not happen with hostpath backend. After some digging I saw that when using ceph backend '/var/lib/kafka/data' is owned by root user instead of kafka and kafka cannot write to it. Thus I think this PR fixes it.

**Pull request merged**